src/hotspot/share/c1/c1_Canonicalizer.cpp

```diff
@@ -469,9 +469,7 @@ void Canonicalizer::do_CompareOp      (CompareOp*       x) {
 
 
 void Canonicalizer::do_IfOp(IfOp* x) {
-  // Caution: do not use do_Op2(x) here for now since
-  //          we map the condition to the op for now!
-  move_const_to_right(x);
+  ShouldNotReachHere();
```

Comment:
Looks good, but maybe add a comment here why we cannot visit `IfOp` here (i.e. that `IfOp` is not created by the `GraphBuilder` but only later when eliminating conditional expressions with `CE_Eliminator`).

Reply:
Updated.

Reply:
Thanks!